backend/src/apiserver/resource/model_converter.go

```diff
@@ -61,45 +61,86 @@ func (r *ResourceManager) ToModelExperiment(inputExperiment interface{}) (*model
+			NumberValue: v2Metric.GetNumberValue(),
+			Format:      v2Metric.GetFormat().String(),
+		}
+	}
+}
+
+func (r *ResourceManager) ToModelRunDetail(runInterface interface{}, runId string, runAt int64, manifest string, templateType template.TemplateType) (*model.RunDetail, error) {
+	switch runInterface.(type) {
+	case *apiv1beta1.Run:
+		return r.ToModelRunDetailV1(runInterface, runId, runAt, manifest, templateType)
+	case *apiv2beta1.Run:
+		return r.ToModelRunDetailV2(runInterface, runId, runAt, manifest, templateType)
+	default:
+		return nil, util.NewInvalidInputError("Invalid Api run type.")
 	}
 }
 
 // The input run might not contain workflowSpecManifest and pipelineSpecManifest, but instead a pipeline ID.
 // The caller would retrieve manifest and pass in.
-func (r *ResourceManager) ToModelRunDetail(run *apiv1beta1.Run, runId string, workflow util.ExecutionSpec, manifest string, templateType template.TemplateType) (*model.RunDetail, error) {
-	resourceReferences, err := r.toModelResourceReferences(runId, common.Run, run.GetResourceReferences())
-	if err != nil {
-		return nil, util.Wrap(err, "Unable to convert resource references.")
-	}
+func (r *ResourceManager) ToModelRunDetailV1(runInterface interface{}, runId string, runAt int64, manifest string, templateType template.TemplateType) (*model.RunDetail, error) {
+	run := runInterface.(*apiv1beta1.Run)
 	var pipelineName string
+	var err error
 	if run.GetPipelineSpec().GetPipelineId() != "" {
 		pipelineName, err = r.getResourceName(common.Pipeline, run.GetPipelineSpec().GetPipelineId())
 		if err != nil {
 			return nil, util.Wrap(err, "Error getting the pipeline name")
 		}
 	}
 
+	// Add a reference to the default experiment if run does not already have a containing experiment
+	ref, err := r.getDefaultExperimentIfNoExperiment(run.GetResourceReferences())
+	if err != nil {
+		return nil, err
+	}
+	if ref != nil {
+		run.ResourceReferences = append(run.GetResourceReferences(), ref)
+	}
+
+	// Convert api ResourceReferences to Model ResourceReferences.
+	resourceReferences, err := r.toModelResourceReferences(runId, common.Run, run.GetResourceReferences())
+	if err != nil {
+		return nil, util.Wrap(err, "Unable to convert resource references.")
+	}
+
 	experimentUUID, err := r.getOwningExperimentUUID(run.ResourceReferences)
 	if err != nil {
 		return nil, util.Wrap(err, "Error getting the experiment UUID")
 	}
 
+	namespace, err := r.getNamespaceFromExperiment(run.GetResourceReferences())
+	if err != nil {
+		return nil, err
+	}
+
 	runDetail := &model.RunDetail{
 		Run: model.Run{
 			UUID:               runId,
 			ExperimentUUID:     experimentUUID,
+			Namespace:          namespace,
+			Name:               run.Name,
 			DisplayName:        run.Name,
-			Name:               workflow.ExecutionName(),
-			Namespace:          workflow.ExecutionNamespace(),
-			ServiceAccount:     workflow.ServiceAccount(),
-			Conditions:         string(workflow.ExecutionStatus().Condition()),
```

Comment:
Yes, since the info here comes from `workflow`, which hasn't been generated yet. It is patched in `updateModelRunWithNewScheduledWorkflow` function [here](https://github.com/kubeflow/pipelines/blob/a294a9d9c147a3761047b0bbdd14b9603aec489e/backend/src/apiserver/resource/model_converter.go#L238).

Reply:
So some fields that used to come from `workflow` now comes from API run directly, but not all of them. Just want to make sure there's no behavior change. 

Would `workflow.ExecutionName()` ever be different from api `v1Run.name`? Similar questions for the rest changes: I guess Namespace won't change, worth double checking, for ServiceAccount, does API run always carry service account info?

Reply:
`Name` and `ServiceAccount` indeed come from API, see [here](https://github.com/kubeflow/pipelines/blob/a294a9d9c147a3761047b0bbdd14b9603aec489e/backend/src/apiserver/template/v2_template.go#L198) and [here](https://github.com/kubeflow/pipelines/blob/a294a9d9c147a3761047b0bbdd14b9603aec489e/backend/src/apiserver/template/v2_template.go#L204). As to namespace, I haven't been able to find where workflow gets its namespace from.

Reply:
I think namespace comes from API as well. See the [current master](https://github.com/kubeflow/pipelines/blob/a4a24885c5ddee1638a36bd9087f10b0a75b0428/backend/src/apiserver/resource/resource_manager.go#L376-L386), `newExecSpec` is created using the `namespace` coming from API.

Reply:
> Name and ServiceAccount indeed come from API, see [here](https://github.com/kubeflow/pipelines/blob/a294a9d9c147a3761047b0bbdd14b9603aec489e/backend/src/apiserver/template/v2_template.go#L198) and [here](https://github.com/kubeflow/pipelines/blob/a294a9d9c147a3761047b0bbdd14b9603aec489e/backend/src/apiserver/template/v2_template.go#L204).

In the referenced code, the assignment flow is executionSpec.* = model.*, and here assignment used to be model.* = workflow.*, and now it's changed to model.* = api.*. My concern is, without digging further into the code, api.name is static, and workflow.name is possibly generated at runtime and different from api.Name (not sure if that's the case, just a guess).
So, let keep digging into the code and make sure there's no behavior change.
@Linchin